### ISSUE TITLE
Add support for stack metadata

### ISSUE THREAD

**Issue by Carles-Figuerola** [bazaarvoice/cloudformation-ruby-dsl#88]
It would be useful to have metadata available for this cloudformation feature:
http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudformation-interface.html

I dug up PR #65 and confirmed that this code:
```
  metadata 'AWS::Cloudformation::Interface' => {
    :ParameterGroups => [ {
      :Label => {
        :default => "App1 Settings"
      },
      :Parameters => [
        :App1InstanceType,
        :App1MinInstanceCount,
        :App1DesiredInstanceCount,
        :App1MaxInstanceCount
      ],
    },
    {
      :Label => {
        :default => "App2 Settings"
      },
      :Parameters => [
        :App2InstanceType,
        :App2MinInstanceCount,
        :App2DesiredInstanceCount,
        :App2MaxInstanceCount
      ]
    } ]
  }
```
produces this json output:
```
{
  "Metadata": {
    "AWS::Cloudformation::Interface": {
      "ParameterGroups": [
        {
          "Label": {
            "default": "App1 Settings"
          },
          "Parameters": [
            "App1InstanceType",
            "App1MinInstanceCount",
            "App1DesiredInstanceCount",
            "App1MaxInstanceCount"
          ]
        },
        {
          "Label": {
            "default": "App2 Settings"
          },
          "Parameters": [
            "App2InstanceType",
            "App2MinInstanceCount",
            "App2DesiredInstanceCount",
            "App2MaxInstanceCount"
          ]
        }
      ]
    }
  }
}
```

**Comment by marjamis** [comment]
+1

**Comment by jonaf** [comment]
Closed by #65
